### PR TITLE
DCOS-14690: Move Environment variables reducer

### PR DESCRIPTION
**Pull request by Poltergeist**
**Checklist**
- [x] Did you add a JIRA issue in a commit message or as part of the branch name?
- [ ] Did you add new unit tests?
- [ ] Did you add new integration tests?
- [ ] If this is a regression, did you write a test to catch this in the future?

part of #2076

### CODE REVIEW

**Comment by wavesoft** [comment]
I have a feeling that something is off with the rebase here :thinking:

**Comment by wavesoft** [comment]
run tests

**Comment by wavesoft** [comment]
retest this please

**Comment by wavesoft** [comment]
Nice, looks good! 👍  ✨

File: plugins/services/src/js/reducers/serviceForm/FormReducers/__tests__/EnvironmentVariables-test.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+      let batch = new Batch();
+      batch = batch.add(new Transaction(['env'], 0, ADD_ITEM));
+      batch = batch.add(new Transaction(['env', 0, 'key'], 'key'));
+      batch = batch.add(new Transaction(['env', 0, 'value'], 'value'));
+
+      expect(batch.reduce(EnvironmentVariables.FormReducer.bind({}), []))
+        .toEqual([{key: 'key', value: 'value'}]);
+    });
+
+    it('should multiple items if they have the same key', function () {
```
Comment (MatApple) [inline plugins/services/src/js/reducers/serviceForm/FormReducers/__tests__/EnvironmentVariables-test.js]: @Poltergeist perhaps this should be `should return multiple...`?

File: plugins/services/src/js/reducers/serviceForm/FormReducers/__tests__/EnvironmentVariables-test.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      batch = batch.add(new Transaction(['env', 1, 'value'], 'value2'));
+
+      expect(batch.reduce(EnvironmentVariables.FormReducer.bind({}), []))
+        .toEqual([
+          {key: 'key', value: 'value'},
+          {key: 'key', value: 'value2'}
+        ]);
+    });
+
+    it('should keep remove the first item', function () {
```
Comment (MatApple) [inline plugins/services/src/js/reducers/serviceForm/FormReducers/__tests__/EnvironmentVariables-test.js]: `keep` or `remove`? Think you meant it to be just remove right :)

**Comment by MatApple** [comment]
Minor #nits otherwise looks good @Poltergeist 👍

**Comment by Poltergeist** [comment]
run integration tests

**Comment by Poltergeist** [comment]
run integration tests

File: plugins/services/src/js/reducers/serviceForm/FormReducers/EnvironmentVariables.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            state = state.filter((item, index) => {
+              return index !== value;
+            });
+            break;
+        }
+
+        return state;
+      }
+
+      const index = joinedPath.match(/\d+/)[0];
```
Comment (mlunoe) [inline plugins/services/src/js/reducers/serviceForm/FormReducers/EnvironmentVariables.js]: We could make this a bit nicer:
```js
const [_base, index, name] = path;
if (type === SET && name === 'key') {
  state[index].key = value;
}
...
```

Comment (Poltergeist) [inline plugins/services/src/js/reducers/serviceForm/FormReducers/EnvironmentVariables.js]: This has turned up in multiple PR's with the same pattern I've created a issue (DCOS-15145) to tackle the problem in all reducers in one go.

**Pull request merged**